media-gfx/converseen/converseen-0.9.11.1.ebuild

```diff
@@ -0,0 +1,67 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake xdg-utils
+
+DESCRIPTION="Batch image converter and resizer based on ImageMagick"
+HOMEPAGE="https://converseen.fasterland.net/
+	https://github.com/Faster3ck/Converseen/"
+SRC_URI="https://github.com/Faster3ck/Converseen/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="debug"
+
+RDEPEND="
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtwidgets:5
+	|| ( media-gfx/imagemagick[cxx] media-gfx/graphicsmagick[cxx,imagemagick] )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	dev-qt/linguist-tools:5
+"
+
+S="${WORKDIR}/${P^}"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-0.9.9.0-appdata-path.patch"
+	"${FILESDIR}/${PN}-0.9.9.0-no-update.patch"
+	"${FILESDIR}/${PN}-0.9.11.1-graphicsmagick-support.patch"
+)
+
+src_prepare() {
+	cmake_src_prepare
+
+	if has_version "media-gfx/graphicsmagick[cxx,imagemagick]"; then
+		# Replace variables in CMakeLists.txt
+		sed -i -e "s/GENTOO_LIB/\/usr\/$(get_libdir)/g" \
+				-e "s/GENTOO_INCLUDE/\/usr\/include/g" \
+				"${S}/CMakeLists.txt" ||
+			die "Failed to sed graphicsmagick patch"
+
+		# Replace MagickCore in globals.cpp
+		sed -i -e "s/MagickCore/MagickLib/" "${S}/src/globals.cpp" ||
```

Comment:
What about something like this in globals.cpp:
```
#if MagickPackageName == GraphicsMagick
const char *version = MagickLib::GetMagickVersion(nullptr);
#else
const char *version = MagickCore::GetMagickVersion(nullptr);
#endif
```

Reply:
With this I fiddled around for quite a while and didn't find a pretty way to compare two strings in a preprocessor instruction.
The version you suggest results in:
```
/usr/include/GraphicsMagick/magick/version.h:36:27: error: token ""GraphicsMagick"" is not valid in preprocessor expressions
   36 | #define MagickPackageName "GraphicsMagick"
```
I read that in preprocessor instructions it's only possible to do integer based comparisons, so string comparisons aren't possible.
I also tried using strcmp for string comparison (in order to get an integer back from strcmp such that I get an integer comparision in the preprocessor instruction), but functions like strcmp aren't allowed either.
I'm still looking for a way to do this without having to define lots of other lengthy macros.

Reply:
Sorry, doh. I think we may want to add an autoconf check for it or something and possibly longer-term add a macro into GM, but that's beyond this for now.